### ISSUE TITLE
Qubes post-Installation fails after reboot in qubes 4.0.2

### ISSUE THREAD

**Issue by Ayush1001** [QubesOS/qubes-issues#5160]
**Qubes OS version**
Qubes 4.0.2

**Affected component(s) or functionality**
laptop is now not booting up stuck on dracut error.

**Brief summary**
After successful installation for qubes 4.0.2 after a simple reboot, unable to load os. Stuck on dracut error. It specify **/dev/mapper/qubes_dom0-root does not exist** & **/dev/qubes_dom0/root does not exist**

**To Reproduce**
Firstly i install qubes 4.0.2 using a 8 gigs usb stick by making bootable using rufus utility. My laptop model is Asus vivobook s15.
I choose legacy support as i was getting error in uefi , and proceed with installation, Installation went fine, and after a reboot all other thing setup also without a errror. But problem came after another reboot. now as laptop want into black screen automatically.
And i select hard-drive from boot menu then it fives me dracut error.  

**Expected behavior**
Should be working fine after a reboot. Os should be loaded without an error.

**Actual behavior**
Stuck on dracut kernal error.
**Screenshots**
I can't provide those a it is related to my office work, but the error is written in above.

**Solutions you've tried**
I tried changing somethings in grub terminal but nothing worked so far.

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**
https://www.qubes-os.org/doc/uefi-troubleshooting/
I had gone through this guide but can't find chainloader..

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**

**Comment by laanwj** [comment]
I had this issue too, today, but in my case the problem was leaving "Create USB qube holding all USB controllers (sys-usb)" enabled while my boot drive was USB. Oops. But it would be nice if it warned or didn't leave this enabled if the boot disk is on the USB controller.

**Comment by notbyosmosis** [comment]
I encountered this issue for the 2nd time this year.  

Check of pool qubes_dom0/pool100 failed (status:!). Manual repair required!
Warning: dracut-initqueue timeout - starting timeout scripts
Warning: Could not boot.
Warning: /dev/mapper/qubes_dom0-root does not exist
Warning: /dev/qubes_dom0/root does not exist

Last time I had to re-install.  Today, through some DuckDuckGo searches, and a small risk, I was able to complete a work-around that got me back up and running.
1. At the Dracut emergency prompt, I ran:

```
lvm 
vgscan
vgchange -ay
```

This informed me that there was an issue with activating qubes_dom0/pool00.
At this point, I tried:

`lvconvert --repair qubes_dom0/pool00`

I received the following error:

```
Read-only locking type set.  Write locks are prohibited.
Can't get lock for qubes_dom0
cannot process volume group qubes_dom0
```

I exited back to the dracut prompt, and temporarily edited the /etc/lvm/lvm.conf file.
I found the following line:

`locking_type=4`

and changed it to:

`locking_type=1`

I then retried the lvconver command:

```
lvm
lvconvert --repair qubes_dom0/pool00
```

It worked!  Recived a couple of warnings.

I then re-ran the following:

```
vgscan
vgchange -ay
```

I then exited back to the dracut prompt and edited the /etc/lvm/lvm.conf file, and changed the locking_type option back to what it was before:

`locking_type=1`

changed it back to:

`locking_type=4`


I then rebooted.  
This time when I put in my luks password, QubesOS came up!

**Comment by john-999** [comment]
5.5 months after this issue was opened you have today saved my life! Without your post, I would have lost 1 year of work!

Without any apparent cause (i.e. no updates applied), QubesOS started to refuse to boot this morning. (It ran perfectly fine yesterday.)

QubesOS is really great, and I admire the people who work on it and I thank them. It's kind of sad that this kind of bug makes me (as an advanced Linux user, but not super-advanced) want to migrate back to Ubuntu. :-(

**Comment by themsay** [comment]
@notbyosmosis, you just add more smile to my new year. Thank you!

**Comment by freewebwithme** [comment]
I got a same issue. And fixed by running lvmconvert.
So what was the reason for this error?

**Comment by mig5** [comment]
@notbyosmosis this just hit me for the first time. Thank you so much for your write-up, it worked perfectly.

**Comment by john-999** [comment]
Today, 4 months later, I just ran into this bug for the 2nd time.

**Comment by vladionescu** [comment]
Our team was just hit by this.

**Comment by Jenn145** [comment]
I'm facing the same issues as OP. I have resolved it once before with the exact steps listed above by member "notbyosmosis". This time after the second use of "lvconvert --repair qubes_dom0/pool00" I'm met with the following errors:

lvm> lvconvert --repair qubes_dom0/pool00

Using default stripesize 64.00 KiB.

WARNING: Sum of all thin volume sizes (1.10 TiB) exceeds the size of thin pools and the size of whole volume group (930.51 GiB)!

For thin pool auto extension activation/thin_pool_autoextend_threshold should be below 100.

Transaction id 4764 from pool "qubes_dom0/pool00" does not match repaired transaction id 4763 from /dev/mapper/qubes_dom0-lvol0_pmspare.

WARNING: recovery of pools without pool metadata spare LV is not automated.

WARNING: If everything works, remove qubes_dom0/pool00_meta1 volume.

WARNING: Use pvmove command to move qubes_dom0/pool00_tmeta on the best fitting PV.

lvm>

(side note.. after completing the repair method the first time the error occured, an black icon/connection in sys-usb appeared and became present in every session. The connection option was named something like: "qubes_dom0/pool00_metadata")

Please help me out if you have any suggestions, my entire OS is unusable with these issues.

**Comment by DemiMarie** [comment]
Culprit appears to be incorrect values in `/etc/lvm/lvm.conf`: `activation/thin_pool_autoextend_threshold` needs to be below 100 to ensure that thin pools are extended when needed.  Futhermore, `activation/error_when_full` needs to be set to `1` (enabled) so that thin pools fail fast if space is exhausted.

**Comment by marmarek** [comment]
> `activation/thin_pool_autoextend_threshold` needs to be below 100 to ensure that thin pools are extended when needed.

Yes. This is actually on my todo-list for R4.1. We already have the installer to leave some free space to have where extend to, just the lvm.conf change is needed.

**Comment by DemiMarie** [comment]
We can change this for existing R4.0 installations, too.

**Comment by marmarek** [comment]
Just change of lvm.conf won't do anything if there is no free space left to actually extend to.

**Comment by DemiMarie** [comment]
True.  I think `error_when_full` is a more important change ― hopefully, it will prevent corruption.

In the long run, a better answer might very well be ZFS Zvol snapshots, which to the best of my understanding do not have this problem.

**Comment by PleaseHelpMeThankYou** [comment]
I am able to complete every step except for the first round of. I cant boot into my OS and my problem is exactly like the one described above. How am I able to edit the file. I tried nano, vim, gedit unless I used the wrong commands idk what else to do and the progress was going so well until this hangup.

I exited back to the dracut prompt, and temporarily edited the /etc/lvm/lvm.conf file.
I found the following line:

locking_type=4

and changed it to:

locking_type=1

I then retried the lvconver command:
